### ISSUE TITLE
Upload binary file possibility in URI module

### ISSUE THREAD

**Issue by mploski** [ansible/ansible-modules-core#3157]
##### Issue Type:
- Feature Idea
##### Plugin Name:

uri
##### Ansible Version:

1.9.2
##### Environment:

N/A
##### Summary:

It would be nice to have option to upload binary file to HTTP endpoint using HTTP PUT. Expecting behaviour similar to following curl command:

```
curl -k -X PUT -u user:password --upload-file  test.jar "https://IP:PORT/endpoint/"
```

and example of possible URI usage:

```
- name: Push jar to nexus
  uri:
    url: "https://{{ inventory_hostname }}:{{ port }}/{{ endpoint }}"
    src: /tmp/{{ file }}.jar
    user: "{{ user }}"
    password: "{{ password }}"
    method: PUT
```

**Comment by ansibot** [comment]
@romeotheriault ping, this issue is waiting for your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@romeotheriault, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by gerhard-tinned** [comment]
+1 

Need this or a similar feature as well. Using ansible 2.1.1.0.

**Comment by wnpaul** [comment]
+1

Need this as well. Using latest ansible for OSX.

**Comment by ansibot** [comment]
@romeotheriault, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by gerhard-tinned** [comment]
What is the new ticket number after the merge??

**Comment by filpamplona** [comment]
+1

Need this or a similar feature as well. Using ansible 2.2.0.

**Comment by gerhard-tinned** [comment]
What is the new ticket number after the merge??
